tests/shotgun_model/test_data_handler_find.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import os
+
+from mock import patch, Mock, call
+from tank_test.tank_test_base import *
+
+# import the test base class
+test_python_path = os.path.abspath(os.path.join( os.path.dirname(__file__), "..", "python"))
+sys.path.append(test_python_path)
+from base_test import TestShotgunUtilsFramework
+
+
+class TestShotgunFindDataHandler(TestShotgunUtilsFramework):
+    """
+    Tests for the data handler low level io
+    """
+    
+    def setUp(self):
+        """
+        Fixtures setup
+        """        
+        super(TestShotgunFindDataHandler, self).setUp()
+        self.shotgun_model = self.framework.import_module("shotgun_model")
+
+    def test_generate_data_request(self):
+        """
+        Test generate_data_request
+        """
+        test_path = os.path.join(self.tank_temp, "test_find_handler.pickle")
+
+        dh = self.shotgun_model.data_handler_find.ShotgunFindDataHandler(
+            entity_type="Asset",
+            filters=[],
+            order=None,
+            hierarchy=["sg_asset_type", "code"],
+            fields=["code"],
+            download_thumbs=True,
+            limit=None,
+            additional_filter_presets=None,
+            cache_path=test_path,
+            parent=None
+        )
+
+        dh.load_cache()
+        self.assertEquals(len(dh.get_entity_ids()), 0)
+
+        # first let the data handler perform its request
+        # and make sure it registers it in an expected way
+        mock_data_retriever = Mock()
+        mock_data_retriever.execute_find = Mock(return_value=1234)
```

Comment:
`Mock` is definitely the coolest Python tool I've discovered since I've started working on Toolkit. I may however be guilty of over-engineering some tests because of it. :)